src/Columns/ColumnVector.cpp

```diff
@@ -437,7 +437,12 @@ template <typename T>
 UInt64 ColumnVector<T>::get64(size_t n [[maybe_unused]]) const
 {
     if constexpr (is_arithmetic_v<T>)
-        return bit_cast<UInt64>(data[n]);
+    {
+        if constexpr (std::endian::native == std::endian::little)
+            return bit_cast<UInt64>(data[n]);
+        else
+            return __builtin_bswap64(bit_cast<UInt64>(data[n]));
+    }
```

Comment:
Looks dangerous because `ColumnVector<T>::get64()` can be used everywhere, not only in datetime calculations

Reply:
This fix should apply for all get64() usage in Big Endian machine. We just happen to caught this error in toDateTime64().

Reply:
fix updated. Removed this change.